File: src/core/src/util/xybuilder/XyBuilderInfo.php
<?php
/**
 * +----------------------------------------------------------------------
 * | uiadmin [ 渐进式快速开发接口后台 ]
 * +----------------------------------------------------------------------
 * | Copyright (c) 2018-2022 http://uiadmin.jiangruyi.com All rights reserved.
 * +----------------------------------------------------------------------
 * | Author: jry <598821125@qq.com>
 * +----------------------------------------------------------------------
*/

namespace uiadmin\core\util\xybuilder;

/**
 * XYBuilderInfo动态信息
 *
 * @author jry <ijry@qq.com>
 */
class XyBuilderInfo {

    // 数据
    private $data;

    /**
     * 初始化
     * @author jry <ijry@qq.com>
     */
    public function init() {
        $this->data = [
            'infoList' => []
        ];
        return $this;
    }

    /**
     * 添加组
     * @author jry <ijry@qq.com>
     */
    public function addInfoGroup(
        $group
    ) {
        $this->data['infoList'][] = $group;
        return $this;
    }

    /**
     * 返回数据
     * @author jry <ijry@qq.com>
     */
    public function getData() {
        foreach ($this->data['infoList'] as $key => &$value) {
            foreach ($value['data'] as $key1 => &$value1) {
                if (!isset($value1['extra'])) {
                    $value1['extra'] = [];
                }
                if (!isset($value1['extra']['sm'])) {
                    $value1['extra']['sm'] = [];
                }
            }
        }
        return $this->data;
    }
}

File: back-end-java/uiadmin-spring-boot/summer-core/src/main/java/com/jiangruyi/summer/core/util/xybuilder/XyBuilderInfo.java
/**
 * 详情页构建
 */
package com.jiangruyi.summer.core.util.xybuilder;

import java.util.*;
import java.io.Serializable;

/**
 * @author jry
 *
 */
public class XyBuilderInfo implements Serializable {
    private static final long serialVersionUID = 1L;

    // 配置
    private Map<String, Object> config = new HashMap<String, Object>();

    // 数据列表
    private List<Object> infoList = new ArrayList<>();

    public XyBuilderInfo() {
        this.config.put("mode", "info");
    }

    // 修改设置
    public Object setConfig(String name, Object value){
        this.config.put(name, value);
        return this;
    }

    // 表单提交方法
    public Object addInfoGroup(Object grpData){
        infoList.add(grpData);
        return this;
    }

    // 获取数据
	public Map<String,Object> getData() {
        HashMap<String, Object> data = new HashMap<String, Object>() {
            {
                put("alertList", new HashMap<String, Object>() {
                    {
                        put("top", new ArrayList<Object>());
                        put("bottom", new ArrayList<Object>());
                    }
                });
                put("infoList", infoList);
                put("config", config);
            }
        };
		return data;
	}
}
